### PR TITLE
feat: listbox text alignment

### PR DESCRIPTION
**Pull request by CostInCode**
Added horizontal text alignment for Listbox

Fixed vertical alignment between frequency count and tick/lock

For testing, pick also the changes in list-objects from https://github.com/qlik-oss/sn-list-objects/pull/29/

https://user-images.githubusercontent.com/13185716/207257584-379b572d-19e9-48c8-b026-30076541c201.mov

Please make sure to read the Pull Request Guidelines:
https://github.com/qlik-oss/nebula.js/blob/master/.github/CONTRIBUTING.md#git
-->

## Requirements checklist

- [ ] Api specification
  - [ ] Ran `yarn spec`
    - [ ] No changes
     ***OR***
    - [ ] API changes has been formally approved
- [ ] Unit/Component test coverage
- [ ] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:
- [ ] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

File: apis/nucleus/src/index.js [hunk 1]
```diff
@@ -419,6 +419,7 @@ function nuked(configuration = {}) {
            * @param {SearchMode=} [options.search=true] Show the search bar permanently or using the toggle button: false|true|toggle|toggleShow
            * @param {boolean=} [options.toolbar=true] Show the toolbar
            * @param {boolean=} [options.checkboxes=false] Show values as checkboxes instead of as fields
+           * @param {string=} [options.testTextAlign="left"] Sets the horizontal text alignment
```
Comment (johanlahti) [inline apis/nucleus/src/index.js:422]: Also specify the possible values for this option, like this:
```
@param {('left'|'center'|'right')} [options.textAlign='left'] Sets the horizontal text alignment.
```

**Comment by johanlahti** [comment]
Also, right before pushing your changes you will have to run `yarn spec` to update the typescript definitions and API docs, which are derived from the jsdocs.

File: apis/nucleus/src/components/listbox/components/__tests__/list-box-row-column.spec.jsx [hunk 2]
```diff
@@ -834,6 +834,30 @@ describe('<ListBoxRowColumn />', () => {
...
+        textAlign,
+      };
+      const testRenderer = await render(
+        <ThemeProvider theme={theme}>
+          <ListBoxRowColumn index={index} style={style} data={data} column={rowCol === 'column'} />
+        </ThemeProvider>
+      );
+      const testInstance = testRenderer.root;
+      const type = testInstance.findByType(Grid);
+      expect(type.props.children[1].props.style.justifyContent).to.equal(textAlign);
```
Comment (johanlahti) [inline apis/nucleus/src/components/listbox/components/__tests__/list-box-row-column.spec.jsx:858]: Best to write the expect as a new string `'right'`, because `textAlign` could have changed on the journey through the component (i.e. if it's not immutable).

File: apis/nucleus/src/components/listbox/components/__tests__/list-box-row-column.spec.jsx [hunk 3]
```diff
@@ -834,6 +834,30 @@ describe('<ListBoxRowColumn />', () => {
...
 
+    it('should get correct text alignment', async () => {
+      const index = 0;
+      const style = {};
+      const textAlign = 'right';
+      const data = {
+        onMouseDown: sandbox.spy(),
+        onMouseUp: sandbox.spy(),
+        onMouseEnter: sandbox.spy(),
+        onClick: sandbox.spy(),
```
Comment (johanlahti) [inline apis/nucleus/src/components/listbox/components/__tests__/list-box-row-column.spec.jsx:845]: Check if we really need to provide these events (since we are not testing them in this test case). Otherwise remove them.

File: apis/nucleus/src/index.js [hunk 4]
```diff
@@ -419,6 +419,7 @@ function nuked(configuration = {}) {
            * @param {SearchMode=} [options.search=true] Show the search bar permanently or using the toggle button: false|true|toggle|toggleShow
            * @param {boolean=} [options.toolbar=true] Show the toolbar
            * @param {boolean=} [options.checkboxes=false] Show values as checkboxes instead of as fields
+           * @param {('left'|'center'|'right')} [options.textAlign='left'] Sets the horizontal text alignment
```
Comment (johanlahti) [inline apis/nucleus/src/index.js:422]: Looking at the `spec.json` generated from the jsdoc it seems we should write it like: `left|center|right` and `options.textAlign=left` (i.e. without quotation mark `'`).

File: apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx [hunk 5]
```diff
@@ -136,6 +136,7 @@ const Root = styled('div')(({ theme }) => ({
       display: 'flex',
       alignItems: 'center',
       paddingLeft: 0,
+      minWidth: '70%', // might need proper calculation
```
Comment (johanlahti) [inline apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx:139]: Why was this needed?

Comment (CostInCode) [reply apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx:139]: to make able the checkbox label span element to occupy the necessary space in order to move the "A" according to justifyContent=textAlign - otherwise it will be always placed on the left 

I will check if I can slightly change it with minWidth and some padding on the right

File: apis/nucleus/src/index.js [hunk 6]
```diff
@@ -422,6 +422,7 @@ function nuked(configuration = {}) {
...
```
Comment (johanlahti) [inline apis/nucleus/src/index.js:425]:
```suggestion
           * @param {(left|center|right)} [options.textAlign='left'] Sets the horizontal text alignment
```

File: apis/nucleus/src/components/listbox/ListBox.jsx [hunk 7]
```diff
@@ -255,6 +255,7 @@ export default function ListBox({
   const dense = layout.layoutOptions?.dense ?? false;
   const { itemSize, listHeight } = getSizeInfo({ isVertical, checkboxes, dense, height });
   const isLocked = layout && layout.qListObject.qDimensionInfo.qLocked;
+  const { textAlign } = layout && layout.qListObject.qDimensionInfo;
```
Comment (johanlahti) [inline apis/nucleus/src/components/listbox/ListBox.jsx:258]: Nebula now supports "optional chaining": `layout?.qListObject.qDimensionInfo`

Comment (johanlahti) [inline apis/nucleus/src/components/listbox/ListBox.jsx:258]:
```suggestion
  const { textAlign } = layout?.qListObject.qDimensionInfo || {};
```

File: apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx [hunk 8]
```diff
@@ -287,6 +292,20 @@ function RowColumn({ index, style, data }) {
 
   const excludedOrAlternative = () => (isAlternative(cell) || isExcluded(cell)) && checkboxes;
 
+  const isNumeric = cell?.qNum !== 'NaN' && cell?.qNum !== undefined;
```
Comment (johanlahti) [inline apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx:295]:
```suggestion
  const isNumeric = !['NaN', undefined].includes(cell?.qNum);
```

File: apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx [hunk 9]
```diff
@@ -287,6 +292,20 @@ function RowColumn({ index, style, data }) {
...
+  let valueTextAlign;
+  const isAutoTextAlign = textAlign === undefined || textAlign.auto;
```
Comment (johanlahti) [inline apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx:297]:
```suggestion
  const isAutoTextAlign = !textAlign || textAlign.auto;
```
(Unless it has be `undefined` precisely, which I don't think it has to be)

File: apis/nucleus/src/components/listbox/components/__tests__/list-box-row-column.test.jsx [hunk 10]
```diff
@@ -827,6 +827,23 @@ describe('<ListBoxRowColumn />', () => {
...
+      const testRenderer = await render(
+        <ThemeProvider theme={theme}>
+          <ListBoxRowColumn index={index} style={style} data={data} column={rowCol === 'column'} />
+        </ThemeProvider>
+      );
+      const testInstance = testRenderer.root;
+      const type = testInstance.findByType(Grid);
+      expect(type.props.children[1].props.style.justifyContent).toEqual('right');
+    });
+
```
Comment (johanlahti) [inline apis/nucleus/src/components/listbox/components/__tests__/list-box-row-column.test.jsx:846]: If possible, it would be great to also add tests for the other cases: align: 'left' and RTL.

**Review by johanlahti — APPROVED** [review]
